File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.go
func findMaxAverage(nums []int, k int) float64 {
    var wSum, mSum float64

    for i := 0; i < k; i++ {
        wSum += float64(nums[i])
    }
    mSum = wSum

    for i := k; i < len(nums); i++ {
        wSum = wSum - float64(nums[i-k]) + float64(nums[i])
        if wSum > mSum {
            mSum = wSum
        }
    }

    return mSum / float64(k)
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {

    let wSum = 0;
    let mSum = -Infinity;

    // initial window
    for (let i = 0; i < k; i++) {
      wSum += nums[i];
    }
    mSum = wSum;

    // slide the window
    for (let i = k; i < nums.length; i++) {
      wSum = wSum - nums[i - k] + nums[i];
      mSum = Math.max(mSum, wSum);
    }

    return mSum/k;
    
};